### ISSUE TITLE
Cursor position is going wrong by a combination of Ctrl+D and Ctrl+C

### ISSUE THREAD

**Issue by kkk669** [holzschu/a-shell#179]
A two types of cursor positions will be out of sync with each other if I press Ctrl+D some times and then press Ctrl+C some times.

## Steps to Reproduce

1. Open a-Shell
2. Clear its screen
3. Press Ctrl+D *N* times
4. Press Ctrl+C *M* times

## Expected Behaviors

The line will be broken *M* times with two cursors overlapping.

## Actual Behaviors

A box cursor will be *N* lines ahead of a vertical bar cursor. The gap will be vanished if the screen is refreshed by user actions e.g. showing or hiding an on-screen keyboard.

https://user-images.githubusercontent.com/601636/107095171-a41afc80-684b-11eb-90d4-c006e2e252cf.MP4

## Environments

I've confirmed that it can be reproduced at least on the following environments.

- a-Shell 1.6.6 and master (d152308505bfdc187e2745dcb283ce19820464a7) on iPadOS 14.4
- a-Shell 1.6.6 on iPadOS 14.5 Beta 1 (18E5140k)

**Comment by holzschu** [comment]
It will be fixed in the next TF release. It's not a generic fix for the two-cursor problem, but it's at least a fix for the control-D problem.

**Comment by kkk669** [comment]
The build 161 works perfectly. Thank you.

**Comment by kkk669** [comment]
This issue happens again at least on a-Shell 1.8.5.

**Comment by holzschu** [comment]
Yes, I can confirm. And the fix introduced in May 2021 is now causing control-D to erase two characters instead of one on the command line.

**Comment by holzschu** [comment]
I have absolutely no idea what is causing this cursor move this time, or how to fix it.

**Comment by holzschu** [comment]
I think I found the solution: https://developer.apple.com/documentation/uikit/uikeycommand/3780513-wantspriorityoversystembehavior 

In the future release, control-D will have this priority.

**Comment by holzschu** [comment]
Version 1.8.6 is now out on the AppStore (not yet on TestFlight), which fixes this issue for iOS 15.
Given the history, I should get ready for a new problem with iOS 16...
